core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -632,86 +655,165 @@ protected void initPreparedStatements() {
                 .concatWith(rxSession.executeAndFetch(findAllMetricsInDataCompressed.bind()));
     }
 
-    @Override
-    public Observable<Integer> insertGaugeData(Metric<Double> gauge) {
-        return insertGaugeData(gauge, -1);
+    /*
+     * Applies micro-batching capabilities by taking advantage of token ranges in the Cassandra
+     */
+    private Observable.Transformer<BoundStatement, Integer> applyMicroBatching() {
+        return tObservable -> tObservable
+                .groupBy(b -> {
+                    ByteBuffer routingKey = b.getRoutingKey(ProtocolVersion.NEWEST_SUPPORTED,
+                            codecRegistry);
+                    Token token = metadata.newToken(routingKey);
+                    for (TokenRange tokenRange : session.getCluster().getMetadata().getTokenRanges()) {
+                        if (tokenRange.contains(token)) {
+                            return tokenRange;
+                        }
+                    }
+                    throw new RuntimeException("Unable to find any Cassandra node to insert token " + token.toString());
```

Comment:
I realize that this situation shouldn't happen, but wouldn't it be better to log a warning and write to any node?